src/engine/plugins/tile.cpp

```diff
@@ -380,226 +387,218 @@ Status TilePlugin::HandleRequest(const std::shared_ptr<datafacade::BaseDataFacad
+                            const auto turn_cost = data.weight - sum_node_weight;
+
+                            // Find the three nodes that make up the turn movement)
+                            const auto node_from = firstnode.first;
+                            const auto node_via = firstedge.target_node;
+                            const auto node_to = secondedge.target_node;
+
+                            const auto coord_from = facade->GetCoordinateOfNode(node_from);
+                            const auto coord_via = facade->GetCoordinateOfNode(node_via);
+                            const auto coord_to = facade->GetCoordinateOfNode(node_to);
+
+                            // Calculate the bearing that we approach the intersection at
+                            const auto angle_in = static_cast<int>(
+                                util::coordinate_calculation::bearing(coord_from, coord_via));
+
+                            // Add the angle to the values table for the vector tile, and get the
+                            // index
+                            // of that value in the table
+                            const auto angle_in_index = use_point_int_value(angle_in);
+
+                            // Calculate the bearing leading away from the intersection
+                            const auto exit_bearing = static_cast<int>(
+                                util::coordinate_calculation::bearing(coord_via, coord_to));
+
+                            // Figure out the angle of the turn
+                            auto turn_angle = exit_bearing - angle_in;
+                            while (turn_angle > 180)
+                            {
+                                turn_angle -= 360;
+                            }
+                            while (turn_angle < -180)
+                            {
+                                turn_angle += 360;
+                            }
+
+                            // Add the turn angle value to the value lookup table for the vector
+                            // tile.
+                            const auto turn_angle_index = use_point_int_value(turn_angle);
+                            // And, same for the actual turn cost value - it goes in the lookup
+                            // table,
+                            // not directly on the feature itself.
+                            const auto turn_cost_index = use_point_float_value(
+                                turn_cost / 10.0); // Note conversion to float here
+
+                            // Save everything we need to later add all the points to the tile.
+                            // We need the coordinate of the intersection, the angle in, the turn
+                            // angle and the turn cost.
+                            all_turn_data.emplace_back(TurnData{
+                                coord_via, angle_in_index, turn_angle_index, turn_cost_index});
```

Comment:
`all_turn_data` contains `TurnData`, correct? If so, `emplace_back` should perfectly forward arguments to its ctor.

Either change the `emplace_back` with the `&&`-taking `push_back` overload here or remove the explicit `TurnData`.